### ISSUE TITLE
cv api doesn't support limit

### ISSUE THREAD

**Issue by twomice** [civicrm/cv#63]
I can't seem to get `cv api` to honor the 'limit' option on any api call. For example, I tried the following, expecting 'count' would be '1', but instead it's '25':

```
✔ /opt/buildkit/build/drupal-demo/web [master ↑·383|✚ 1…3] 
12:42 $ cv api Contact.get sequential=1 options={"limit":1}  | grep '"count"'
    "count": 25,
✔ /opt/buildkit/build/drupal-demo/web [master ↑·383|✚ 1…3] 
12:42 $ cv --version
cv version v0.3.1
```
Perhaps worth noting:
* This `cv` command is copied from the API Explorer. Perhaps it's wrong?
* I'm using the latest `cv` version downloaded with `curl -LsS https://download.civicrm.org/cv/cv.phar -o /opt/buildkit/bin/cv && chmod u+x /opt/buildkit/bin/cv`

Should I maybe try a different format for the `options` array?

**Comment by MegaphoneJon** [comment]
I don't think using JSON in `cv` like this works.  The way I do this is:
`echo <my API command> | cv api:batch`

**Comment by totten** [comment]
Historically...

* The initial command was `cv api`, and it accepted JSON input like 
    ```bash
    echo '{api-params}' | cv api --in=json
    ```
* `cv api:batch` was added to support multiple command invocations (bidirectional pipes). Here, it's assumed/defined that all input is piped JSON.
    ```bash
    echo '[[api-entity, api-action, api-params], ... ]' | cv api:batch
    ```
* `cv api4` was a copy of `cv api` which defaulted to `version: 4`. However, in early usage, I got quickly frustrated with typing the commands to pipe JSON on CLI,  so the input parser was expanded to support "JSON-ish Key-Value". That still wasn't ergonomic enough, so then I also added `+Options` (and I'm happier with that). The following are equivalent but use different input features.
    ```bash
    ## Using pure JSON pipe; same format as `cv api`
    echo '{"select":["id","display_name"],"limit":10}' | cv api4 Contact.get --in=json

    ## Using JSON-ish Key-Value
    ## Gist: Starts out doing "Key=Value" for alphanumeric keys/values; but
    ## for anything fancy... use JSON characters (`{`, `[`, `"`, `:`).
    cv api4 Contact.get limit=10 select='["id","display_name"]'

    ## Using +Options
    cv api4 Contact.get +s id,display_name +l 10
    ```

If you want something that's mechanically-simple, correct, and always behaves the same (*e.g. for data integrations*) -- then use `--in=json`. But if you want something more ergonomic (*for CLI interaction*), then use +Options or JSON-ish Key-Value.

The `cv api4 --help` screen describes this stuff. One can experiment with the notations using the dry-run/verbose options (`cv api4 -Nv`):

![Screen Shot 2020-02-26 at 1 16 16 PM](https://user-images.githubusercontent.com/1336047/75388656-38d65580-589a-11ea-8b2d-4623e0c5930b.png)

So why is that "JSON-ish Key-Value" was added to `cv api4` but *not* `cv api`? It'd break some APIv3 users. "Basic Key-Value" and "JSON-ish Key-Value" behave differently if the key or value begins with a reserved character.

That's not a common/typical issue... but it's also not theoretical. Here's a practical example. Consider a command to set `userFrameworkResourceURL` to `[cms.root]/libraries/civicrm` on build of `roundearth/drupal-civicrm-project`:

```bash
## Correct usage in Basic Key-Value 
## ... because `[` is a string-literal
cv api setting.create 'userFrameworkResourceURL=[cms.root]/libraries/civicrm'

## Incorrect usage in JSON-ish Key-Value
## ... because `[` switches us into JSON
cv api4 Setting.create 'userFrameworkResourceURL=[cms.root]/libraries/civicrm'

## Correct usage in JSON-ish Key-Value
## ... because `"` switches us into JSON
cv api4 Setting.create 'userFrameworkResourceURL="[cms.root]/libraries/civicrm"'
```

Anyway, this means that the path of least resistance was to let `cv api` and `cv api4` have slightly different contracts. I'm not opposed to trying to reconcile the contracts... but we need someone to say it seems worthwhile...

**Comment by totten** [comment]
tldr: if API Explorer suggested using that with `cv api`, then I'd call it a bug in API Explorer - or else we need to figure out a breaking-change in `cv api`.

**Comment by twomice** [comment]
Great explanation. Thanks @totten .
